### DISCUSSION TITLE
A few requests

### DISCUSSION THREAD

**Discussion by andy-thomason** [DataEngineeringLabs/arrow2-convert#109] [Ideas]
I have used arrow files and indexed databases in genetics research for many years.

It is especially useful to have schemas with nested List of Struct components which this derive
supports for serialization quite well, but doing random reads from record batches would also
be useful. Memory mapping makes arrow files very powerful and we used terabyte sized
files to store indexed genetic variant data.

It would be useful to have a third "ref" mode where we build structs of array references:

```
#[derive(ArrowRef)]
struct MyAOS {
    other: Vec<OtherAOS>
}

// generating
struct MyAOSRef<'a> {
   other: &'a ListArray<i32>,
}
```
   
Dictionaries are also useful and the MutableDictionaryArray wrapper makes
it easy to build them. We can custom build ArrowSerialize wrappers for primitives which do this
but an attribute might be worth it.

**Comment by ncpenke** [comment]
Sorry @andy-thomason, I somehow missed this. This is a great idea, and will see if it can be done!
